File: javascript/acronym/acronym.js
export const parse = (phrase) => {
    const acronym = [];
    const re = /([a-z'])[a-z']*/gi;
    let match;
    while ((match = re.exec(phrase)) !== null) {
        acronym.push(match[1].toUpperCase());
    }
    return acronym.join('');
};

File: clojure/acronym/src/acronym.clj
(ns acronym
  (:require [clojure.string :refer [join upper-case]]))

(defn acronym
  "Converts phrase to its acronym."
  [phrase]
  (->> phrase
       (re-seq #"([A-Za-z])[A-Z]*[a-z]*")
       (map second)
       (map upper-case)
       join))
